File: app/src/main/java/social/entourage/android/user/UserModule.java
package social.entourage.android.user;

import dagger.Module;
import dagger.Provides;

/**
 * Module related to UserFragment
 * @see UserFragment
 */
@Module
public class UserModule {
    private final UserFragment fragment;

    public UserModule(final UserFragment fragment) {
        this.fragment = fragment;
    }

    @Provides
    public UserFragment providesUserFragment() {
        return fragment;
    }
}

File: app/src/main/java/social/entourage/android/user/UserModule.kt
package social.entourage.android.user

import dagger.Module
import dagger.Provides

/**
 * Module related to UserFragment
 * @see UserFragment
 */
@Module
class UserModule(private val fragment: UserFragment) {
    @Provides
    fun providesUserFragment(): UserFragment {
        return fragment
    }

}